peparser.cpp

```diff
@@ -1323,38 +1323,49 @@ namespace peparser
 		return false;
 	}
 
-	bool PEParser::DetectMIDLMarker(size_t diffStart, size_t diffSize, size_t& diffShift) const
+	bool PEParser::DetectMIDLMarker(size_t diffStart, size_t diffSize, size_t& diffShift, bool tlbCmpExpr) const
 	{
-		diffShift = 0;
+        diffShift = 0;
 
-		UsefulBlockMapRange markers = m_useful.equal_range(MidlTimestampSegment);
-		if(markers.first == markers.second) return false;
+        if (!tlbCmpExpr)
+            return false;
 
-		Block marker;
+        const auto it = std::find_if(cbegin(m_resourceBlocks), cend(m_resourceBlocks), [](const Block& block) {
+            return std::wstring::npos != block.description.find(L"@TYPELIB");
+        });
 
-		for(UsefulBlockMap::const_iterator it = markers.first; it != markers.second; ++it)
-		{
-			if(diffStart < it->second.offset || diffStart >= it->second.offset + it->second.size)
-				continue;
+        if (cend(m_resourceBlocks) != it)
+        {
+            const std::string magic = "Created by MIDL version";
 
-			marker = it->second;
-			break;
-		}
+            const char* stringStart = FindStringEntry< char, None >(
+                (char*)m_view + it->offset, magic, magic.size(), it->size);
 
-		if(marker.offset == 0) 
-			return false;
+            if (stringStart == nullptr)
+                return false;
 
-		const char* sectionStart = (char*)((LPBYTE)m_view + marker.offset);
+            const struct
+            {
+                const unsigned char byte;
+                const size_t offset;
 
-		std::string magic = "Created by MIDL version";
-		const char* stringStart = FindStringEntry<char, None>(sectionStart, magic, magic.size(), marker.size);
+                bool check(const void* data) const noexcept {
+                    return data && byte == *((unsigned char *)data + offset);
+                }
+            }
+            lf{ 0x0a, 61 }, dc3{ 0x13, 62 };
 
-		if(stringStart == NULL) 
-			return false;
+            if (lf.check(stringStart) && dc3.check(stringStart))
+            {
+                const size_t tlbStampLength = 65;
+                const auto diffPos = (char*)m_view + diffStart;
 
-		diffShift = marker.size - diffStart + marker.offset;
+                if (stringStart < diffPos && diffPos < stringStart + tlbStampLength)
+                    return true;
```

Comment:
ok (I was copied the diffShift variable usage from a DetectFILEMacro function).
And what is diffShift means? It a difference (in bytes) of ignored block size?